aten/src/ATen/native/cpu/DistributionTemplates.h

```diff
@@ -147,6 +147,50 @@ static void normal_fill_16(scalar_t *data, const scalar_t mean, const scalar_t s
     data[j + 8] = radius * std::sin(theta) * std + mean;
   }
 }
+ 
+#if defined(__VSX__)  || defined(CPU_CAPABILITY_VSX)
+static void normal_fill_16_VSX(float *data,Vectorized<float> &two_pi,Vectorized<float> &one,Vectorized<float> &minus_two, Vectorized<float> &mean,Vectorized<float> &std) {
+  using Vec = Vectorized<float>;
+  Vec u1=one-Vec::loadu(data);// [0, 1) -> (0, 1] for log.
+  Vec u2=Vec::loadu(data+Vec::size());
+  Vec radius=(minus_two * u1.log());//std::sqrt(-2 * std::log(u1));
+  radius=radius.sqrt();
+  Vec theta=two_pi * u2;
+  Vec output_vec=radius * theta.cos() * std + mean;
+  Vec output_vec2=radius * theta.sin() * std + mean;
+  output_vec.store(data);
+  output_vec2.store(data+Vec::size());
+}
+
+template <typename RNG>
+void normal_fill_VSX(const TensorBase &self, const float mean, const float std, RNG generator) {
+  float *data = self.data_ptr<float>();
+  auto   size = self.numel();
+  std::lock_guard<std::mutex> lock(generator->mutex_);
+  for (const auto i : c10::irange(size)) {
+    at::uniform_real_distribution<float> uniform(0, 1);
+    data[i] = uniform(generator);
+  }
+
+  using Vec = Vectorized<float>;
+  Vec two_pi = Vec(2.0f * c10::pi<double>);
+  Vec one = Vec(1.0f);
+  Vec minus_two = Vec(-2.0f);
+  Vec var_vec  = Vec(std);
+  Vec mean_vec = Vec(mean);
+  for (int64_t i = 0; i < size - 15; i += 16) {
+    normal_fill_16_VSX(data + i, two_pi, one, minus_two, mean_vec, var_vec);
+  }
+  if (size % 16 != 0) {
+    data = data + size - 16;
+    for (const auto i : c10::irange(16)) {
+      at::uniform_real_distribution<float> uniform(0, 1);
+      data[i] = uniform(generator);
+    }
+     normal_fill_16_VSX(data, two_pi, one, minus_two, mean_vec, var_vec);
+  }
+}
```

Comment:
The implementation calls the general vectorized abstraction without invoking the intrinsics directly. Can we make it a general vectorized implementation for all archs?

Reply:
Yes

Reply:
Can you address my question: https://github.com/pytorch/pytorch/pull/113351/files#r1395037569?